DESIGN.md

```diff
@@ -325,18 +325,99 @@ In the addition of the previous rules, an operator for `Flowable`:
+
+(*The unfortunate effect of `create` in 1.x was the ignorance of the Observable contract and beginner's first choice as an entry point. We can't eliminate this path since `rs.Publisher` is a single method functional interface that can be implemented just as badly.*)
+
+Therefore, new standard factory methods will try to address the common entry point requirements: 
+  - `create(SyncOnSubscribe)` to safe, synchronous generation of signals, one-by-one
+  - `create(AsyncOnSubscribe)` to batch-create signals based on request patterns
+  - `create(Consumer<? super SingleEmitter<T>>)` to relay a single value or error from other reactive sources (i.e., addListener callbacks)
+  -  `create(Consumer<? super FlowEmitter<T>>)` to relay multiple values or error from multi-valued reactive-sources (i.e., button-clicks) while also give flow control options right there (buffer, drop, error, etc.).
+
+The first two `create` methods take an implementation of an interface which provides state and the generator methods:
 
 ```java
-Flowable.create(SyncGenerator generator)
+interface SyncOnSubscribe<T, S> {
+
+    S createState();
+    
+    S generate(S state, Observer<T> output);
+    
+    void disposeState(S state);
+}
+
+interface AsyncOnSubscribe<T, S> {
+
+    S createState();
+    
+    S generate(S state, long requested, Observer<Observable<T>> output);
+    
+    void disposeState(S state);
+}
+```
+
+These latter two `create` methods will provide the following interaction interfaces to the `java.util.function.Consumer`:
+
+```java
+interface SingleEmitter<T> {
+
+    complete(T value);
+    
+    fail(Throwable error);
+    
+    stop();
+    
+    setDisposable(Disposable d);
+    
+}
+
+interface FlowEmitter<T> {
+
+    void next(T value);
+   
+    void fail(Throwable error);
+   
+    void complete();
+   
+    void stop();
 
-Flowable.create(AsyncGenerator generator)
+    setDisposable(Disposable d);
 
-Observable<T>.create(OnSubscribe<Observer<T>> onSubscribe)
+    enum BackpressureHandling {
+        IGNORE,
+        ERROR,
+        DROP,
+        LATEST,
+        BUFFER
+    }
 
-Single<T>.create(OnSubscribe<Single.Subscriber<T>> onSubscribe)
+    void setBackpressureHandling(BackpressureHandling mode);
+    
+}
+```
+
+By extending the base classes, operator implementations would loose the tracking/wrapping features of 1.x. To avoid this, the methods `subscribe(C)` will be final and operators have to implement a protected `subscribeActual` (or any other reasonable name).
+
+```java
+@Override
+public final void subscribe(Subscriber<? super T> s) {
+   subscribeActual(hook.onSubscribe(s));
+} 
 
-Completable<T>.create(OnSubscribe<Completable.Subscriber<T>> onSubscribe)
+protected abstract void subscribeActual(Subscriber<? super T> s);
+```
+
+Assembly-time hooks will be moved into the individual standard methods on the base types:
```

Comment:
There has been no previous mention of the hooks in this document. Is it necessary to include this in the design?